### DISCUSSION TITLE
Update all VM in Netbox - netbox_virtual_machine

### DISCUSSION THREAD

**Discussion by Sampaiolc** [netbox-community/ansible_modules#749] [Q&A]
Hello, 

I would like to update all VMs on my Netbox. The problem is that the ansible module "netbox_virtual_machine" works only locally(localhost). So how do you generally update Netbox? 

Do you have any examples of usage?

**Comment by sc68cal** [comment]
If you are using the [netbox inventory](https://netbox-ansible-collection.readthedocs.io/en/latest/getting_started/how-to-use/inventory.html) plugin, you can build an inventory file for all your VMs and write playbooks that way, or you can use `nb_lookup` as a `loop:` parameter for your tasks where you want to set things in `netbox_virtual_machine` tasks.

**Reply by Sampaiolc** [reply]
In https://docs.ansible.com/ansible/latest/collections/netbox/netbox/netbox_virtual_machine_module.html say:

"Note
    Tags should be defined as a YAML list
    This should be ran with connection local and hosts localhost"
    
I can run "netbox_virtual_machine" only in localhost. The facts in ansible get only recurses about my AWX(Ansible machine).

**Reply by sc68cal** [reply]
If you are trying to do `netbox_virtual_machine` calls with facts, then yes you will have to use `nb_inventory` plugin to access facts that are gathered. However it is not a hard requirement that you _must_ use `connection: local` or `hosts: localhost`
